### PR TITLE
Powershell process keep running with local train transport.

### PR DESCRIPTION
**Pull request by Vasu1105**
Signed-off-by: Vasu1105 <vasundhara.jagdale@chef.io>

## Description

While running compliance remediation cookbook for Windows Server 2019 one of the customer has reported that its creating multiple Powershell processes as defined in issue #358 after debugging the same.

This PR adds the implementation of close method which kills all running processes once the execution of the command is done.
Issue can be reproduce with following
```
cmds = ["dir", "mkdir test", "hostname", "$env:PATH"]
cmds.each do |cmd|
  conn = Train.create('local').connection
  conn = conn.run_command(cmd)
 end
```
Testing close method

```
cmds = ["dir", "mkdir test", "hostname", "$env:PATH"]
cmds.each do |cmd|
  conn = Train.create('local').connection
  conn = conn.run_command(cmd)
  conn.close
 end
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New content (non-breaking change)
- [ ] Breaking change (a content change which would break existing functionality or processes)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,27 +1,24 @@
-<!-- latest_release 3.4.7 -->
-## [v3.4.7](https://github.com/inspec/train/tree/v3.4.7) (2021-01-11)
-
-#### Merged Pull Requests
-- Test on ruby 3.0, drop testing on ruby 2.4 [#657](https://github.com/inspec/train/pull/657) ([clintoncwolfe](https://github.com/clintoncwolfe))
+<!-- latest_release -->
```
Comment (clintoncwolfe) [inline CHANGELOG.md:1]: Changes to the CHANGELOG should happen automaticaly when the PR merges - including them on a PR will only introduce conflicts.  These changes should not be included.

File: lib/train/transports/local.rb [hunk 2]
```diff
@@ -160,12 +174,22 @@ def run_command(cmd)
...
         def acquire_pipe
+          require "win32/process"
+
           pipe_name = "inspec_#{SecureRandom.hex}"
 
-          start_pipe_server(pipe_name)
+          @server_pid = start_pipe_server(pipe_name)
+
+          # Ensure process is killed when the Train process exits
+          at_exit { Process.kill("KILL", @server_pid) unless @server_pid.nil? }
```
Comment (mwrock) [inline lib/train/transports/local.rb:192]: my ruby is rusty but could you call `self.close` instead of repeating the `Process.kill` here

**Review by mwrock — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Vasu1105) [reply CHANGELOG.md:1]: Definitely this got pushed by mistake will revert this one. Thanks @clintoncwolfe

File: lib/train/transports/local.rb (hunk 2 above)
Comment (Vasu1105) [reply lib/train/transports/local.rb]: Thanks @mattray.  Did the required changes.

Comment (tecracer-theinen) [reply lib/train/transports/local.rb]: I hit that exact line of code yesterday for another reason. Maybe we can include the fix to that here as well?

When programmatically using InSpec from a custom Chef resource, the broken pipe of the KILL signal results in a failed Chef run as the `at_exit` will make the run exit non-zero on exceptions. I had a simple `rescue true` suffix for the `Process.kill` command at hand.

Question only remains if this PR should be merged first and I hand in my addition of if we append the fix here.

**Comment by clintoncwolfe** [comment]
It looks like the added test is failing the linter - run `bundle exec chefstyle -a` to fix.

File: lib/train/transports/local.rb (hunk 2 above)
Comment (clintoncwolfe) [reply lib/train/transports/local.rb]: At this point, we're only killing the subprocess if we know the PID - that is, we believe that the process is running. So I think that should cover the common cases. I think absorbing all exceptions will likely hide new problems. I'm reluctant to add a catch-all rescue unless we must.

**Review by clintoncwolfe — APPROVED** [review]
Thanks!

**Pull request merged**